examples/utils/bundle-size-tests/src/debugAssert.ts

```diff
@@ -0,0 +1,14 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { debugAssert } from "@fluidframework/core-utils/internal";
+
+// TODO: ideally there would be a unit test which actually checks that this is omitted from production builds.
+// For now it can be manually verified in ../dist/debugAssert.js,
+// and any regression breaking it will show up as a bundle size regression.
+debugAssert(() => "This should be removed in production");
```

Comment:
This produces a tiny bundle:

```Javascript
var bundle;(()=>{"use strict";throw new Error("This should be kept")})();
//# sourceMappingURL=debugAssert.js.map
```

This confirms that even the entire debugAssert call sight, debugAssert function and all its dependencies are omitted in their entirety.

Reply:
Theoretically, we could give this test its own dir + build config and we could check in the resulting `.js` file as a "snapshot". We could also go a step further and make a "real" snapshot test that asserts the contents don't change. Would probably be good to automatically detect a regression here.